storm-server/src/main/java/org/apache/storm/blobstore/BlobStoreUtils.java

```diff
@@ -191,6 +192,8 @@ public static boolean downloadUpdatedBlob(Map<String, Object> conf, BlobStore bl
                     out.close();
                 }
                 isSuccess = true;
+            } catch(FileNotFoundException fnf) {
+                LOG.warn("FileNotFoundException", fnf);
```

Comment:
rethrow fnf?

Reply:
It crashes the nimbus if the exception is propagated. (we have seen this happening if topology gets killed while the non-leader nimbus tries to download the blob). It may be better to swallow and let `downloadUpdatedBlob` return false here?

Reply:
Line 271 of this file is already dealing with the race. so in theory eating the FileNotFound or rethrowing it as a KeyNotFound exception should be fine.  I am just trying to understand where the FileNotFound is coming from.

The remoteBlobStore should only be getting thrift exceptions.  The only place in the code I see that could throw a FileNotFoundException is when we are interacting with the local blob, but even then the blobstore code should be changing it into a KeyNotFound or some other thrift exception.

Reply:
I see. but then, any other `IOException` such as failure to reach Blobstore etc could cause nimbus restart. I would treat all IOExceptions in same manner. i.e. log them and return `false`.

Reply:
Okay I think I see it now.  On the local file output stream it can show up if the file/directory is deleted before we do the commit.  It would be good to document this, preferably in the LOG message, because FileNotFoundException is redundant, it is already going to log that and knowing why this is only a warning is preferable.